core/src/lib.rs

```diff
@@ -760,6 +761,60 @@ pub fn export_drawing_to_disk(
     )
 }
 
+#[derive(Debug, Serialize, EnumIter)]
+pub enum ImportFileError {
+    NoAccount,
+    ParentDoesNotExist,
+    FileAlreadyExists,
+    DocumentTreatedAsFolder,
+    BadPath,
+}
+
+pub fn import_file(
+    config: &Config,
+    parent: Uuid,
+    source: PathBuf,
+    edit: bool,
+    import_progress: Option<Box<dyn Fn(ImportExportFileInfo)>>,
+) -> Result<(), Error<ImportFileError>> {
+    import_export_service::import_file(config, parent, source, edit, import_progress).map_err(|e| {
+        match e {
+            CoreError::AccountNonexistent => UiError(ImportFileError::NoAccount),
+            CoreError::FileNonexistent => UiError(ImportFileError::ParentDoesNotExist),
+            CoreError::FileNotFolder => UiError(ImportFileError::DocumentTreatedAsFolder),
+            CoreError::DiskPathInvalid => UiError(ImportFileError::BadPath),
+            CoreError::PathTaken => UiError(ImportFileError::FileAlreadyExists),
+            _ => unexpected!("{:#?}", e),
+        }
+    })
+}
+
+#[derive(Debug, Serialize, EnumIter)]
+pub enum ExportFileError {
+    NoAccount,
+    ParentDoesNotExist,
+    FileAlreadyExistsInDisk,
+    BadPath,
```

Comment:
I don't want to be too nit picky, but the CoreError names for these (`DiskPathTaken` and `DiskPathInvalid`) genuinely seem much clearer to me. Disregard this comment if you disagree, no big deal.

Reply:
agreed

Reply:
I agree.